synapse/push/baserules.py

```diff
@@ -238,6 +238,25 @@ def make_base_prepend_rules(
         ],
         "actions": ["dont_notify"],
     },
+    # Enable notifications for replies without fallback
```

Comment:
Instead of adding default rules I wonder if its better to leave them out, and if clients want to test them they can add them itself?

Reply:
How about defaulting the default rule to disabled, if the experimental option in the HS config isn't set? Clients can still add their own push rules that way or decide to enable it and the HS admin can enable it globally?

Reply:
TBH, I'm not sure what the effect of that would be for clients? I vaguely recall that adding new default push rules is a bit of a PITA, but I can't really remember why. Plus not sure how clients would react to these new push rules (it's probably fine, but I'm not 100% sure)

What about only including them if the experimental option is set?

Reply:
So if you don't enable it without the option set, it shouldn't affect notifications, because it isn't enabled. The major issue I could foresee, is that some clients would fail to parse the rule? I'm not sure how strict push rule parsers in clients are, but I would expect them to just ignore unknown conditions in disabled rules.

If you don't include the rule at all, clients can't know, if the condition type is supported. That might be fine and we could expose a flag in /capabilities instead. That option has a bit more implementation overhead, but that might be worth it for the safety? I can look into it this weekend.

Reply:
> I vaguely recall that adding new default push rules is a bit of a PITA, but I can't really remember why. Plus not sure how clients would react to these new push rules (it's probably fine, but I'm not 100% sure)

Related: https://github.com/matrix-org/matrix-doc/pull/3664#discussion_r789520216